### PR TITLE
[Master] Restructuring frontend and backend

### PR DESCRIPTION
**Pull request by wangyyt1013**
## Summary

This PR restructures backend and frontend functionalities into separate folders, and connects them through setting up proxy in 'client/package.json` and appropriate route handlers in 'server/server.js'.

## Proposed changes
1. Moved all of the contents of the react app into `client` folder. Added proxy configuration into its `package.json`.
2. Based on PR #50 , set up backend logic in the `server` folder so that it accepts requests forwarded from frontend.
3. Based on PR #50 , set up backend routes and API skeleton code for further implementation.

## Dependencies
This PR depends on backend logic changes proposed by PR #50, which is only a WIP PR to serve an example for code reviews and not for merging purposes. It's also based off the most recent status of the `master` branch, so please be cautious about the potential merge conflicts it might have with the `frontend` branch.

## Linked issue
Resolves issue #46 

## Reviewers
ptal:
- Databases team: @TannerTreasure @justinbchen @ic372 
- EMs from other teams: @erin2722 @Paul2858 @aassal @CarolineHoang

### CODE REVIEW

File: server/routes/clubProfileRoutes.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const fns = require('../controllers/clubProfileController')
```
Comment (TannerTreasure) [inline server/routes/clubProfileRoutes.js:1]: change the name of fns to something like functions so this code is a bit more clear/easier to read

File: server/server.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+})
+
+// parsing AJAX requests
+app.use(bodyParser.urlencoded({extended: true}))
+app.use(bodyParser.json())
+
+// routes
+app.use('/api/clubProfiles', clubProfileRoutes)
+
+app.use(express.static(path.join(__dirname, '../client/build')))
```
Comment (TannerTreasure) [inline server/server.js:21]: 'best practice'

**Review by TannerTreasure — COMMENTED** [review]
everything makes sense to me

File: client/package.json [hunk 3]
```diff
@@ -2,6 +2,7 @@
   "name": "clubs-2020",
   "version": "0.1.0",
   "private": true,
+  "proxy": "http://localhost:8080",
```
Comment (erin2722) [inline client/package.json:5]: ive never done this before! Does this mean that our react app runs on port 8080 or that our react app on port 3000 will be able to perform fetch requests to port 8080?

**Review by erin2722 — COMMENTED** [review]
looks good 👍

File: client/package.json (hunk 3 above)
Comment (wangyyt1013) [reply client/package.json:5]: yes! all the fetch calls sent by the frontend (listening on localhost:3000) will be directed to the backend server (listening on localhost:8080)

**Pull request merged**